### PR TITLE
Writer review for THORN-2122

### PR DESCRIPTION
**Pull request by tradej**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

Notes:

* Basically the same as Ladicek#1, plus:
* I've indented the appropriate lines in the `prometheus.yml` config file by 2 spaces to correspond to the actual output from the command.

### CODE REVIEW

File: docs/howto/metrics/index.adoc [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 :parent-context: {context}
 
+{Thorntail} provides ways of measuring the performance of your application and reacting to service outages.
```
Comment (Ladicek) [inline docs/howto/metrics/index.adoc]: This is a somewhat inacurate summary. I'd say "Thorntail provides ways of exposing application metrics", or "Thorntail provides ways of exposing application metrics in order to track performance and service availability", or something like that.

File: docs/howto/metrics/proc_exposing-metrics.adoc [hunk 2]
```diff
@@ -2,95 +2,109 @@
 [id='exposing-metrics_{context}']
 = Exposing metrics
 
+Exposing metrics to Prometheus--a monitoring solution--can be done in a configuration file in your {Thorntail}-based application.
+Collect the metrics using annotations in your application source code.
```
Comment (Ladicek) [inline docs/howto/metrics/proc_exposing-metrics.adoc]: These 2 lines are inacurate and probably duplicate the "what is metrics" chapter.

File: docs/howto/metrics/proc_exposing-metrics.adoc [hunk 3]
```diff
@@ -2,95 +2,109 @@
...
+
+In this example, you:
+
+* Configure your application to collect metrics.
```
Comment (Ladicek) [inline docs/howto/metrics/proc_exposing-metrics.adoc]: Not collect but expose. Prometheus is who collects them.

File: docs/howto/metrics/proc_exposing-metrics.adoc [hunk 4]
```diff
@@ -2,95 +2,109 @@
...
+* View the data using Prometheus.
```
Comment (Ladicek) [inline docs/howto/metrics/proc_exposing-metrics.adoc]: Collect and view.

File: docs/howto/metrics/proc_exposing-metrics.adoc [hunk 5]
```diff
@@ -2,95 +2,109 @@
...
 ----
-
-By default, Prometheus expects metrics to be exposed on `/metrics`.
-This can be configured in `prometheus.yml`, but you will notice that we didn't have to configure anything like that.
-This is because MicroProfile Metrics specifies that metrics are exposed on `/metrics`, and that's exactly what {Thorntail} does.
-
-This configuration snippet also shows an important characteristic of Prometheus: it actively connects to monitored applications to collect data.
-It doesn't expect applications to actively send metrics to some server.
++
+Prometheus expects metrics to be exposed at the `/metrics` endpoint of your application, which is a requirement of the MicroProfile Metrics specification.
```
Comment (Ladicek) [inline docs/howto/metrics/proc_exposing-metrics.adoc]: This changes meaning a bit too much to my taste. What is important is that Prometheus and MP Metrics agree on a common URL.

File: docs/howto/metrics/proc_exposing-metrics.adoc [hunk 6]
```diff
@@ -2,95 +2,109 @@
...
 [source,xml]
 ----
 <dependencies>
 include::pom.xml[tag=mp-metrics,indent=2]
 </dependencies>
 ----
---
 
-. Annotate the desired methods or classes with the desired metrics annotations.
+. Annotate methods or classes with the metrics annotations that you want, for example:
```
Comment (Ladicek) [inline docs/howto/metrics/proc_exposing-metrics.adoc]: It's unclear what part of the sentence the "that you want" bit binds to.

File: docs/howto/metrics/prometheus.yml [hunk 7]
```diff
@@ -18,20 +18,20 @@ rule_files:
...
 # the text above is a verbatim copy of /etc/prometheus/prometheus.yml from the prom/prometheus image:
 # docker run -it --rm --entrypoint /bin/sh prom/prometheus -c 'cat /etc/prometheus/prometheus.yml'
 
 # tag::thorntail[]
-- job_name: 'thorntail'
-  static_configs:
-  - targets: ['localhost:8080']
+  - job_name: 'thorntail'
+    static_configs:
+    - targets: ['localhost:8080']
```
Comment (Ladicek) [inline docs/howto/metrics/prometheus.yml:36]: I would prefer not changing this file, as it's a verbatim copy of the default Prometheus config.

File: docs/howto/metrics/ref_metrics.adoc [hunk 8]
```diff
@@ -2,14 +2,23 @@
...
-This fraction is everything that's required on the application side.
-Outside of the application, it is expected that Prometheus will be running in the environment.
-In this example, we'll run Prometheus locally, using Docker.
-One popular solution to visualize metrics stored in Prometheus is link:https://grafana.com/[Grafana], but we won't discuss it here.
+In the microservices architecture, where multiple services are invoked in order to serve a single user request, diagnosing performance issues or reacting to service outages might be hard.
+To make solving problems easier, applications must expose machine-readable data about their behavior, among others:
+
+* How many requests are currently being processed.
+* How many connections to the database are currently in use.
+* How long service invocations take.
```
Comment (Ladicek) [inline docs/howto/metrics/ref_metrics.adoc]: Those were just examples, not some particularly important types of metrics.

File: docs/howto/metrics/ref_metrics.adoc [hunk 9]
```diff
@@ -2,14 +2,23 @@
...
+
+{Thorntail} provides a fraction for MicroProfile Metrics, an easy-to-use API for exposing metrics.
+Among other formats, it supports exporting data in the native format of Prometheus, a popular monitoring solution.
+Inside the application, you need nothing except this fraction.
+Outside of the application, Prometheus typically runs.
+
+.Additional resources
+* The link:https://github.com/eclipse/microprofile-metrics[MicroProfile Metrics GitHub page^].
+* The link:https://prometheus.io/[Prometheus homepage^]
+* A popular solution to visualize metrics stored in Prometheus is Grafana. For more information, see the link:https://grafana.com/[Grafana homepage^].
```
Comment (Ladicek) [inline docs/howto/metrics/ref_metrics.adoc]: Two sentences on a single line :-)

File: docs/howto/metrics/index.adoc (hunk 1 above)
Comment (tradej) [reply docs/howto/metrics/index.adoc]: OK.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 2 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: OK, I'll come up with a better intro.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 3 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: OK.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 4 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: OK.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 5 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: Was either of them first or does either of them conform to the other's specification?

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 6 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: OK, will probably remove.

File: docs/howto/metrics/prometheus.yml (hunk 7 above)
Comment (tradej) [reply docs/howto/metrics/prometheus.yml:36]: When I run the command to produce the config, it's different from what you have in the original PR.

File: docs/howto/metrics/ref_metrics.adoc (hunk 8 above)
Comment (tradej) [reply docs/howto/metrics/ref_metrics.adoc]: They are meant as such, and they're arranged in a list to catch the eye, along the lines of "What are metrics?" "Look at this list of examples." It's a recommended technique.

File: docs/howto/metrics/ref_metrics.adoc (hunk 9 above)
Comment (tradej) [reply docs/howto/metrics/ref_metrics.adoc]: Impeccable accuracy. :)

File: docs/howto/metrics/prometheus.yml (hunk 7 above)
Comment (tradej) [reply docs/howto/metrics/prometheus.yml:36]: I've now actually piped the output directly in the `prometheus.yml` file, and I've missed two more comments originally. The `prometheus.yml` file in 5a0fbfd should be 100% accurate now.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 6 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: I have removed the part because it works without it too.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 5 above)
Comment (Ladicek) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: I believe it's MP Metrics conforming to what Prometheus expects, but not totally sure.

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 2 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: I have removed this part as I decided it was completely unnecessary.

File: docs/howto/metrics/prometheus.yml (hunk 7 above)
Comment (Ladicek) [reply docs/howto/metrics/prometheus.yml:36]: Ah maybe IntelliJ IDEA reformatted the text when I pasted it into an editor...

File: docs/howto/metrics/proc_exposing-metrics.adoc (hunk 5 above)
Comment (tradej) [reply docs/howto/metrics/proc_exposing-metrics.adoc]: OK, thanks. I will try to find a neutral wording.

File: docs/howto/metrics/ref_metrics.adoc (hunk 8 above)
Comment (Ladicek) [reply docs/howto/metrics/ref_metrics.adoc]: Would it be possible to replace "among others" with "such as"? I think that would better convey the intent that it's just examples.

Comment (tradej) [reply docs/howto/metrics/ref_metrics.adoc]: Sure, no problem.

**Comment by tradej** [comment]
I hope I've addressed all feedback. If so, feel free to squash and merge. (I've left the commits separate for easier review.)
